### ISSUE TITLE
Existing posts are not shown in the profile

### ISSUE THREAD

**Issue by leo-unglaub** [bluesky-social/social-app#3301]
Hey,
i am building a bsky client and for that is was playing around with your API. Creating Posts (com.atproto.repo.createRecord) and so on. On the API site it works fine, but i found a bug in your web interface. When i create multiple posts that are identical (same `text` and `createdAt` property) then only one i shown in the web version. However, if you delete the shown post, all of them are not shown anymore. But the posts still exist (checked via API) and also via profile counter.

![Screenshot_2024-03-21_00-49-47](https://github.com/bluesky-social/social-app/assets/734684/42e58e61-73fd-46b4-a0e1-9d6494df001e)

As you see, there are 11 posts, but none are shown. The number 11 is confirmed via API call.

![image](https://github.com/bluesky-social/social-app/assets/734684/f3473484-e240-4da9-b390-9e996fc2d797)
![image](https://github.com/bluesky-social/social-app/assets/734684/82229bf9-59cd-44a2-ae4d-358ea2897017)

Greetings
Leo

**Comment by leo-unglaub** [comment]
Ah yes, here is the json from that API response. In case you need to copy yhe uri/cid.

```json
{
  "uri": "at://did:plc:acv4ytdlpduxjnsfcqxrd524/app.bsky.feed.post/3ko664amp6k2i",
  "cid": "bafyreih7gmv3qhjqlx5epvkzrqeqsmuj7fy7tgkxe52md6zxaaqoojsgdi",
  "value": {
    "text": "API Test",
    "$type": "app.bsky.feed.post",
    "createdAt": "2024-03-21T00:44:10Z"
  }
}
```

**Comment by leo-unglaub** [comment]
Addition: Just checked, it is possible to find then via the search on your site. But they are still missing from the profile.
![image](https://github.com/bluesky-social/social-app/assets/734684/ea85fc5a-09cd-461f-a7b2-4094d2d252ce)

**Comment by mary-ext** [comment]
Not really seeing anything on your repository at the moment, no posts are there.
![image](https://github.com/bluesky-social/social-app/assets/148872143/98fcd87d-b39f-4123-946e-0cb6c595b452)

**Comment by mary-ext** [comment]
> ```json
> at://did:plc:acv4ytdlpduxjnsfcqxrd524/app.bsky.feed.post/3ko664amp6k2i
> ```

![image](https://github.com/bluesky-social/social-app/assets/148872143/63c802f5-5c5c-4ab5-ad1f-484dd44fd3e1)

**Comment by leo-unglaub** [comment]
Yes, i deleted them. I wanted to check if deleting posts has an effect on the post count. But it did not. 

But since reproducing this is very easy, me deleting this should not affect your ability to investigate this.
